src/transforms/branch.ts

```diff
@@ -37,31 +32,34 @@ async function getLastCommit(github: GitHubAPI, webhookPayload: WebhookPayloadCr
 		url,
 		updateSequenceId: Date.now()
 	};
-}
+};
 
-export const transformBranch = async (github: GitHubAPI, webhookPayload: WebhookPayloadCreate) => {
-	if (webhookPayload.ref_type !== "branch") return undefined;
+export const transformBranch = async (github: GitHubAPI, webhookPayload: WebhookPayloadCreate, jiraHost: string, installationId: number, logger: LoggerWithTarget): Promise<JiraBranchData | undefined> => {
+	if (webhookPayload.ref_type !== "branch") {
+		return;
+	}
 
 	const { ref, repository } = webhookPayload;
-
 	const issueKeys = issueKeyParser().parse(ref) || [];
 
-	if (_.isEmpty(issueKeys)) {
-		return undefined;
+	if (isEmpty(issueKeys)) {
+		return;
 	}
 
-	const lastCommit = await getLastCommit(github, webhookPayload, issueKeys);
-
-	const newPrUrl = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_PULL_REQUEST_URL_FORMAT, false);
+	const gitHubClient = new GitHubClient(getCloudInstallationId(installationId), logger);
+	const [useNewPrUrl, useNewGitHubClient] = await Promise.all([
+		booleanFlag(BooleanFlags.USE_NEW_GITHUB_PULL_REQUEST_URL_FORMAT, false, jiraHost), 
+		booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_BRANCH_EVENT, false, jiraHost)
+	]);
+	const lastCommit = await getLastCommit((useNewGitHubClient ? gitHubClient : github), webhookPayload, issueKeys);
```

Comment:
I went inline with the ternary, thought it would be a bit silly to make a throw away const after the changes above 😂
will be ripped out in clean up too....